cli/_cli.php

```diff
@@ -104,3 +145,68 @@ function validateOptions(array $input, array $params): bool {
 	fwrite(STDERR, sprintf("FreshRSS error: unknown options: %s\n", implode (', ', $unknownOptions)));
 	return false;
 }
+
+/**
+ * Checks for use of unknown parameters with FreshRSS' CLI commands.
+ * @param array<string> $input An array of parameters to check for validity.
+ * @param array<string> $params An array of valid parameters to check against.
+ * @return array<string> Returns an array of all unknown parameters found.
+ */
+function findInvalidOptions(array $input, array $params): array {
+	$sanitizeInput = getLongOptions($input, REGEX_INPUT_OPTIONS);
+	$unknownOptions = array_diff($sanitizeInput, $params);
+
+	if (0 === count($unknownOptions)) {
+		return [];
+	}
+
+	fwrite(STDERR, sprintf("FreshRSS error: unknown options: %s\n", implode (', ', $unknownOptions)));
+	return $unknownOptions;
+}
+
+/**
+ * Checks for use of deprecated parameters with FreshRSS' CLI commands.
+ * @param array<string> $options User inputs to check for deprecated parameter use.
+ * @param array<string,string> $params An array with replacement parameters as keys and their respective deprecated
+ * parameters as values.
+ * @return bool Returns TRUE and generates a deprecation warning if deprecated parameters
+ * have been used, FALSE otherwise.
+ */
+function checkforDeprecatedParameterUse(array $options, array $params): bool {
+	$deprecatedOptions = array_intersect($options, $params);
+	$replacements = array_map(static fn($option) => array_search($option, $params, true), $deprecatedOptions);
+
+	if (0 === count($deprecatedOptions)) {
+		return false;
+	}
+
+	trigger_error("The FreshRss CLI option(s): " . implode (', ', $deprecatedOptions) .
```

Comment:
Although the native function `trigger_error(..., E_USER_DEPRECATED)` is nice, it generates some outputs, which I found a bit misleading. Users could tend to believe that is is related to some internal code, due to the `PHP Deprecated: ` prefix, and ` in /var/www/FreshRSS/cli/_cli.php on line 183` suffix (which I would like to strip).

```
PHP Deprecated:  The FreshRss CLI option(s): base_url, default_user are deprecated and will be removed in a future release. Use: base-url, default-user instead in /var/www/FreshRSS/cli/_cli.php on line 183
```

I think we should use a normal STDERR output like we do for some other messages.

Reply:
That all makes sense. I'll switch it over.